addons/website_sale_loyalty/controllers/delivery.py

```diff
@@ -7,36 +7,31 @@
 
 class WebsiteSaleLoyaltyDelivery(WebsiteSaleDelivery):
 
-    def _update_website_sale_delivery_return(self, order, **post):
+    def _order_summary_values(self, order):
         if order:
```

Comment:
No, it is not in the super. if no order it returns `{}`

Reply:
Indeed, but does it really make sense for `shop_update_carrier` to be called when there is no cart ?
I believe we should do nothing if there is no cart (or raise), and call `_order_summary_values` only if there is a cart :thinking:.

Also, it could be nice to avoid calling `_update_programs_and_rewards` when the carrier did not effectively change (maybe in an override of `_set_carrier` instead ?)